src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/OneVersusAllTrainer.cs

```diff
@@ -499,6 +505,10 @@ private abstract class ImplBase : ISingleCanSavePfa
             public abstract ValueMapper<VBuffer<float>, VBuffer<float>> GetMapper();
             public abstract JToken SaveAsPfa(BoundPfaContext ctx, JToken input);
 
+            public bool CanSaveOnnx(OnnxContext ctx) => Predictors.All(pred => (pred as ICanSaveOnnx)?.CanSaveOnnx(ctx) == true);
```

Comment:
>(pred as ICanSaveOnnx)? [](start = 79, length = 23)

What happens if it cant cast?

Reply:
If it can't cast, it will return a null. And the question mark after the ')' ensures makes it a null conditional operator and will not execute further and the overall function will return false.

---
In reply to: [348211703](https://github.com/dotnet/machinelearning/pull/4462#discussion_r348211703) [](ancestors = 348211703)